stream.go

```diff
@@ -103,28 +110,66 @@ func (stream *Stream) connect() (r io.ReadCloser, err error) {
 	return
 }
 
+func (stream *Stream) backoffWithJitter(attempts int) time.Duration {
+	retry := stream.retry.Nanoseconds()
+	max := stream.maxReconnectionTime.Nanoseconds()
+
+	exp := pow(2, attempts)
+
+	jitterVal := retry * int64(exp)
+
+	if exp == 0 || jitterVal > max || jitterVal <= 0 {
+		jitterVal = max
+	}
+
+	return time.Duration(jitterVal/2 + rand.Int63n(jitterVal)/2)
+}
+
+// Integer power: compute a**b, from Knuth
+func pow(a, b int) int {
```

Comment:
Strictly speaking, `b` should be uint, as this is only correct for non-negative exponents